src/gsy_e/models/market/two_sided.py

```diff
@@ -442,16 +440,30 @@ def _replace_orders_with_residual_in_recommendations_list(
         replaced with corresponding residual offer/bid
         """
 
-        def replace_recommendations_with_residuals(recommendation: Dict):
+        def _replace_matching_requirements(recommendation):
+            if "energy" in getattr(recommendation, "matching_requirements", {}
+                                   ).get("bid_requirement", {}):
+                for index, requirement in enumerate(recommendation["bid"]["requirements"]):
+                    if requirement == recommendation["matching_requirements"]["bid_requirement"]:
+                        bid_requirement = deepcopy(requirement)
+                        bid_requirement["energy"] -= bid_trade.traded_energy
+                        recommendation["bid"]["requirements"][index] = bid_requirement
+                        recommendation["matching_requirements"][
+                            "bid_requirement"] = bid_requirement
+                        return recommendation
+            return recommendation
+
+        def _replace_recommendations_with_residuals(recommendation: Dict):
             if (recommendation["offer"]["id"] == offer_trade.offer_bid.id and
                     offer_trade.residual is not None):
                 recommendation["offer"] = offer_trade.residual.serializable_dict()
             if (recommendation["bid"]["id"] == bid_trade.offer_bid.id and
                     bid_trade.residual is not None):
                 recommendation["bid"] = bid_trade.residual.serializable_dict()
+                recommendation = _replace_matching_requirements(recommendation)
```

Comment:
why is this only added to the bid case?

Reply:
This is because at the moment the `energy` requirement is only supported in the bid cases, so offers can't really have residual requirements